src/core/mac/mac_frame.cpp

```diff
@@ -1112,9 +1112,12 @@ Frame::InfoString Frame::ToInfoString(void) const
 BeaconPayload::InfoString BeaconPayload::ToInfoString(void) const
 {
     const uint8_t *xpanid = GetExtendedPanId();
+    otNetworkName  networkname;
+
+    strlcpy(networkname.m8, GetNetworkName(), sizeof(networkname.m8));
 
     return InfoString("name:%s, xpanid:%02x%02x%02x%02x%02x%02x%02x%02x, id:%d ver:%d, joinable:%s, native:%s",
-                      GetNetworkName(), xpanid[0], xpanid[1], xpanid[2], xpanid[3], xpanid[4], xpanid[5], xpanid[6],
+                      networkname.m8, xpanid[0], xpanid[1], xpanid[2], xpanid[3], xpanid[4], xpanid[5], xpanid[6],
```

Comment:
I think we do ensure the network name (in `Mac`) is NULL terminated.
Here are the two `SetNetworkName()`. Am I missing something? 
````
otError Mac::SetNetworkName(const char *aNetworkName)
{
    return SetNetworkName(aNetworkName, OT_NETWORK_NAME_MAX_SIZE + 1);
}

otError Mac::SetNetworkName(const char *aBuffer, uint8_t aLength)
{
    otError error  = OT_ERROR_NONE;
    uint8_t newLen = static_cast<uint8_t>(strnlen(aBuffer, aLength));

    VerifyOrExit(newLen <= OT_NETWORK_NAME_MAX_SIZE, error = OT_ERROR_INVALID_ARGS);
    VerifyOrExit(newLen != strlen(mNetworkName.m8) || memcmp(mNetworkName.m8, aBuffer, newLen) != 0,
                 Get<Notifier>().SignalIfFirst(OT_CHANGED_THREAD_NETWORK_NAME));

    memcpy(mNetworkName.m8, aBuffer, newLen);
    mNetworkName.m8[newLen] = 0;
    Get<Notifier>().Signal(OT_CHANGED_THREAD_NETWORK_NAME);

exit:
    return error;
}
````

Reply:
Sorry! Got it.... This is not the `Mac` one. The `Frame` one.